java/code/src/com/redhat/rhn/frontend/xmlrpc/system/provisioning/powermanagement/PowerManagementHandler.java

```diff
@@ -0,0 +1,403 @@
+            server = SystemManager.lookupByIdAndUser(serverId.longValue(), loggedInUser);
+        }
+        catch (LookupException e) {
+            throw new NoSuchSystemException();
+        }
+        CobblerPowerSettingsUpdateCommand cmd = new CobblerPowerSettingsUpdateCommand(
+                loggedInUser, server,
+                data.get(PowerManagementAction.POWER_TYPE),
+                data.get(PowerManagementAction.POWER_ADDRESS),
+                data.get(PowerManagementAction.POWER_USERNAME),
+                data.get(PowerManagementAction.POWER_PASSWORD),
+                data.get(PowerManagementAction.POWER_ID));
+        ValidatorError error = cmd.store();
+        if (error != null) {
+            throw new InvalidParameterException(error.getMessage());
+        }
+
+        return 1;
+    }
+
+    /**
+     * Set power management settings for the given system
+     *
+     * @param loggedInUser the user
+     * @param nameIn the cobbler name prefix
+     * @param data power management parameters
+     * @return current power management settings when available
+     *
+     * @xmlrpc.doc Get current power management settings of the given system
+     * @xmlrpc.param #param("string", "sessionKey")
+     * @xmlrpc.param #param("string", "name")
+     * @xmlrpc.param
+     *  #struct_begin("data")
+     *    #prop_desc("string", "powerType", "Power management type")
+     *    #prop_desc("string", "powerAddress", "IP address for power management")
+     *    #prop_desc("string", "powerUsername", "The Username")
+     *    #prop_desc("string", "powerPassword", "The Password")
+     *    #prop_desc("string", "powerId", "Identifier")
+     *  #struct_end()
+     * @xmlrpc.returntype #return_int_success()
+     */
+    public int setDetails(User loggedInUser, String nameIn, Map<String, String> data) {
+        ensureOrgAdmin(loggedInUser);
+        CobblerUnregisteredPowerSettingsUpdateCommand cmd =
+                new CobblerUnregisteredPowerSettingsUpdateCommand(
+                loggedInUser, nameIn,
+                data.get(PowerManagementAction.POWER_TYPE),
+                data.get(PowerManagementAction.POWER_ADDRESS),
+                data.get(PowerManagementAction.POWER_USERNAME),
+                data.get(PowerManagementAction.POWER_PASSWORD),
+                data.get(PowerManagementAction.POWER_ID));
+        ValidatorError error = cmd.store();
+        if (error != null) {
+            throw new InvalidParameterException(error.getMessage());
+        }
+
+        return 1;
+    }
+
+    /**
+     * Return a list of available power management actions
+     *
+     * @param loggedInUser the user
+     * @return list of action strings
+     *
+     * @xmlrpc.doc Return a list of available power management actions
+     * @xmlrpc.param #param("string", "sessionKey")
+     * @xmlrpc.returntype #array_single("string", "power management actions")
+     */
+    public List<String> listActions(User loggedInUser) {
```

Comment:
in case `doPowerManagement` is split up this function would not be needed i guess.

Reply:
yes, done